domains/authorization/Authorization.API/API/Authorization/[Features]/GetUserOrganizationConsentsReceivedQuery.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Repository;
+using API.ValueObjects;
+using EnergyOrigin.Domain.ValueObjects;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Authorization._Features_;
+
+public class GetUserOrganizationConsentsReceivedQueryHandler(IOrganizationConsentRepository organizationConsentRepository)
+    : IRequestHandler<GetUserOrganizationConsentsReceivedQuery, GetUserOrganizationConsentsReceivedQueryResult>
+{
+    public async Task<GetUserOrganizationConsentsReceivedQueryResult> Handle(GetUserOrganizationConsentsReceivedQuery request, CancellationToken cancellationToken)
+    {
+        var userIdpUserIdClaim = IdpUserId.Create(Guid.Parse(request.IdpUserId));
+        var userOrgCvrClaim = Tin.Create(request.OrgCvr);
+
+        var consents = await organizationConsentRepository
+            .Query()
+            .Where(consent => consent.ConsentReceiverOrganization.Tin == userOrgCvrClaim &&
+                              consent.ConsentReceiverOrganization.Affiliations
+                                  .Any(o => o.User.IdpUserId == userIdpUserIdClaim))
+            .Select(consent =>
+                new GetUserOrganizationConsentsReceivedQueryResultItem(
+                    consent.Id,
+                    consent.ConsentGiverOrganizationId,
+                    consent.ConsentGiverOrganization.Tin!.Value,
+                    consent.ConsentGiverOrganization.Name.Value,
+                    UnixTimestamp.Create(consent.ConsentDate).EpochSeconds)
+            )
+            .ToListAsync(cancellationToken: cancellationToken);
```

Comment:
Du søger på consent.Id, og derefter på consent.ConsentGiverOrganizationId. er det nødvendigt? linje 29-30

Reply:
I don't follow?!? -  line 27- 34 is a projection, the filtering happens in line 24-26 in the where clause.

Reply:
ah. det er min fejl